### PR TITLE
Add section about SSH signing keys to developer docs.

### PR DESCRIPTION
**Pull request by alliepiper**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,10 +33,33 @@ Thank you for your interest in contributing to the CUDA C++ Core Libraries (CCCL
...
    ```
 
+### SSH Signing Keys
+
+CCCL projects require commits to be signed with a cryptographic key.
+To enable commit signing using your existing ssh key, set the following git options:
+
+```bash
+git config --global gpg.format ssh
+git config --global user.signingKey "$(cat ~/.ssh/id_rsa.pub)"
```
Comment (miscco) [inline CONTRIBUTING.md:50]: Nitpick: I would change that to
```suggestion
git config --global user.signingKey "$(cat Location_Of_Your_Public_Key)"
```

**Review by miscco — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -33,10 +33,33 @@ Thank you for your interest in contributing to the CUDA C++ Core Libraries (CCCL
...
+
+   Setup ssh signing keys (if needed, see next section) and commit:
+
    ```bash
    git commit -m "Brief description of the change"
    ```
 
+### SSH Signing Keys
+
+CCCL projects require commits to be signed with a cryptographic key.
```
Comment (jrhemstad) [inline CONTRIBUTING.md:45]: comment: This isn't quite true. Signed commits are only necessary for internal contributors who want CI to run automatically and not require `/ok to test`.

**Review by jrhemstad — CHANGES_REQUESTED** [review]
Requesting changes for now as I'm on the fence about putting this in the contributing guide. I don't want to overwhelm new contributors with a bunch of stuff that it seems like they need to do. Especially with this one where it strictly isn't really necessary. 

If we do add it, I'd prefer to put in the CI details section here: https://github.com/NVIDIA/cccl/blob/main/ci-overview.md#ci-workflow-details

File: CONTRIBUTING.md (hunk 2 above)
Comment (alliepiper) [reply CONTRIBUTING.md:45]: Ah, ok, I'd misunderstood that nuance. I'll gladly move it to the ci docs, as long as I have something concise to link (and refer back to myself 😄)

File: ci-overview.md [hunk 3]
```diff
@@ -95,6 +95,28 @@ CCCL uses [NVIDIA's self-hosted action runners](https://docs.gha-runners.nvidia.
 
 The CI pipeline will not start automatically for external contributors. A repository member will first review the changes and initiate the CI pipeline with an `/ok to test` comment.
 
+### SSH Signing Keys
+
+NVIDIA employees on the CCCL team must enable commit signing in order to run CI themselves.
+This is not required for external contributions, which will always require an explicit `/ok to test` comment from an approved account for each CI run.
+To enable commit signing using your existing ssh key, set the following git options:
```
Comment (jrhemstad) [inline ci-overview.md:102]:
```suggestion
[Signed commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) are required for any internal NVIDIA contributors who want the convenience of CI running automatically whenever a commit is pushed to a branch (i.e., doesn't require using `/ok to test`). 

This is not required for external contributions, which will always require an explicit `/ok to test` comment from an approved account for each CI run.

To enable commit signing using your existing ssh key, set the following git options:
```

File: ci-overview.md [hunk 4]
```diff
@@ -95,6 +95,28 @@ CCCL uses [NVIDIA's self-hosted action runners](https://docs.gha-runners.nvidia.
...
+
+```bash
+git config --global gpg.format ssh
+git config --global user.signingKey "$(cat ~/.ssh/YOUR_PUBLIC_KEY_FILE_HERE.pub)"
```
Comment (jrhemstad) [inline ci-overview.md:106]: You can just specify the path. 
```suggestion
git config --global user.signingKey  ~/.ssh/YOUR_PUBLIC_KEY_FILE_HERE.pub
```

File: ci-overview.md [hunk 5]
```diff
@@ -95,6 +95,28 @@ CCCL uses [NVIDIA's self-hosted action runners](https://docs.gha-runners.nvidia.
...
+# If these are set to false, use `git commit -S` to manually sign each commit.
+git config --global commit.gpgsign true
+git config --global tag.gpgsign true
+```
+
+Git is now configured to sign commits with your ssh key.
+
+To complete the process, upload the public key to your [Github Signing Keys](https://github.com/settings/keys).
+Make sure that the key is uploaded to 'Signing Keys', not just 'Authentication Keys'.
+The same key may be used for both.
```
Comment (jrhemstad) [inline ci-overview.md:118]:
```suggestion

To complete the process, upload the public key to your [Github Signing Keys](https://github.com/settings/keys) using the `gh` CLI:
```
gh ssh-key add ~/.ssh/YOUR_PUBLIC_KEY_FILE_HERE.pub --type signing
```

```

**Review by jrhemstad — APPROVED** [review]

**Pull request merged**